namespace-registry/registry.go

```diff
@@ -0,0 +1,411 @@
+    clientNonce, _ := data["client_nonce"].(string)
+	serverNonce, err := generateNonce()
+	if err != nil {
+		log.Errorln("Error generating nonce")
+		ctx.JSON(500, gin.H{"error": "Failed to generate nonce for key sign challenge"})
+		return
+	}
+
+    serverPayload := []byte(clientNonce + serverNonce)
+
+	privateKey, err := loadServerKeys()
+	if err != nil {
+		log.Warningln("Failure to load the server's private key:", err)
+		ctx.JSON(500, gin.H{"error": "Server is unable to generate a key sign challenge"})
+		return
+	}
+
+    serverSignature, err := signPayload(serverPayload, privateKey)
+	if err != nil {
+		log.Warningln("Failure when signing the challenge:", err)
+		ctx.JSON(500, gin.H{"error": "Failure when signing the challenge"})
+		return
+	}
+	ctx.JSON(http.StatusOK, gin.H{
+        "server_nonce": serverNonce,
+        "client_nonce": clientNonce,
+        "server_payload": hex.EncodeToString(serverPayload),
+        "server_signature": hex.EncodeToString(serverSignature),
+    })
+}
+
+func jwksToEcdsaPublicKey(jwks map[string]interface{}) *ecdsa.PublicKey {
+	x := jwks["x"].(string)
+	y := jwks["y"].(string)
+	xBigInt, _ := new(big.Int).SetString(x, 10)
+	yBigInt, _ := new(big.Int).SetString(y, 10)
+
+	clientPubkey := &ecdsa.PublicKey{
+		Curve: elliptic.P521(),
+		X:     xBigInt,
+		Y:     yBigInt,
+	}
+
+	return clientPubkey
+}	
+
+func keySignChallengeCommit(ctx *gin.Context, data map[string]interface{}, action string) {
+    clientNonce, _ := data["client_nonce"].(string)
+    serverNonce, _ := data["server_nonce"].(string)
+    jsonPublicKey := data["pubkey"].(map[string]interface{})
+
+	clientPubkey := jwksToEcdsaPublicKey(jsonPublicKey)
+	clientPayload := []byte(clientNonce + serverNonce)
+	clientSignature, err := hex.DecodeString(data["client_signature"].(string))
+	if err != nil {
+		log.Warningln("Failed to decode 'client_signature' value:", err)
+		ctx.JSON(500, gin.H{"error": "Failed to decode 'client_signature' value"})
+		return
+	}
+	clientVerified := verifySignature(clientPayload, clientSignature, clientPubkey)
+
+	serverPayload, _ := hex.DecodeString(data["server_payload"].(string))
+	serverSignature, _ := hex.DecodeString(data["server_signature"].(string))
+	serverPrivateKey, err := loadServerKeys()
+	if err != nil {
+		log.Warningln("Failed to load server private key:", err)
+		ctx.JSON(500, gin.H{"error": "Failed to load server private key"})
+		return
+	}
+	serverPubkey := serverPrivateKey.PublicKey
+	serverVerified := verifySignature(serverPayload, serverSignature, &serverPubkey)
+
+    if clientVerified && serverVerified {
+        if action == "register" {
+			log.Debug("Registering namespace", data["prefix"])
+			dbAddNamespace(ctx, data)
+        } 
+    } else {
+        ctx.JSON(http.StatusMultipleChoices, gin.H{"status": "Key Sign Challenge FAILED"})
+    }
+}
+
+/*
+Handler functions called upon by the gin router
+*/
+func cliRegisterNamespace(c *gin.Context) {
```

Comment:
Rename from `c` to `ctx`; using `c` makes it hard to search for the variable in text editors.